src/test/java/com/mabl/integration/jenkins/MablStepDeploymentRunnerTest.java

```diff
@@ -187,6 +188,60 @@ public void continueOnPlanFailure() throws IOException, MablSystemError {
         verify(client).close();
     }
 
+    @Test
+    public void planWithRetrySuccess() throws IOException, MablSystemError {
+        MablStepDeploymentRunner runner = new MablStepDeploymentRunner(
+                client,
+                outputStream,
+                TEST_POLLING_INTERVAL_MILLISECONDS,
+                environmentId,
+                applicationId,
+                labels,
+                false,
+                false,
+                true,
+                buildPath,
+                envVars
+        );
+
+        when(client.createDeploymentEvent(eq(environmentId), eq(applicationId), eq(labels), any(CreateDeploymentProperties.class)))
+                .thenReturn(new CreateDeploymentResult(eventId, "workspace-w"));
+
+        when(client.getExecutionResults(eventId))
+                .thenReturn(createExecutionResultWithRetry(true, true));
```

Comment:
Should the success on first attempt be false here?

Reply:
Good catch. Thank you.